client/src/components/JobList.tsx

```diff
@@ -20,15 +36,50 @@ export default function JobList() {
     loadJobs();
   }, []);
 
+  /**
+   * This block filters the jobs list based on the inputs from FilterForm.tsx
+   * There is a ternary for each input field.
+   * Salary ternary handles the number to string comparison;
+   *   The select options for salary increments by 50000, hence line 55
+   */
+  const filteredJobs = jobs?.filter(
+    (item) =>
+      (searchInput
+        ? item.title.toLowerCase().includes(searchInput.toLowerCase()) ||
+          item.companyName.toLowerCase().includes(searchInput.toLowerCase())
+        : true) &&
+      (locationSelect ? item.location === locationSelect : true) &&
+      (levelSelect ? item.level === levelSelect : true) &&
+      (salarySelect
+        ? item.salaryFloor >= Number(salarySelect) &&
+          item.salaryFloor < Number(salarySelect) + 50000
+        : true)
+  );
+
+  /**
+   * These varaibles determine the number of pages needed based on 'filteredJobs"
+   * That data is passed to Pagination.tsx
+   */
   const lastIndex = currentPage * itemsPerPage;
   const firstIndex = lastIndex - itemsPerPage;
-  const currentJobs = jobs?.slice(firstIndex, lastIndex);
+  const currentJobs = filteredJobs?.slice(firstIndex, lastIndex);
```

Comment:
The only reason currentJobs and filteredJobs can be undefined is because jobs can be. Fix that and this all simplifies.